functorch/_src/aot_autograd.py

```diff
@@ -906,6 +927,22 @@ def aot_module_simplified(mod: nn.Module, *top_args, **top_kwargs) -> nn.Module:
     params_flat = tuple(params_flat)
     params_len = len(params_flat)
 
+    shape_env = _produce_or_verify_shape_env(top_kwargs["shape_env"])
+    fake_mode = FakeTensorMode(shape_env=shape_env) if config.use_fake_tensor else nullcontext()
```

Comment:
I feel like you should probably be propagating the fake tensor mode down here too

Reply:
Oh is this the double fake thing you were talking about